packages/mui-base/src/SelectUnstyled/useSelect.types.ts

```diff
@@ -41,14 +41,20 @@ interface UseSelectCommonProps<TValue> {
 export interface UseSelectSingleParameters<TValue> extends UseSelectCommonProps<TValue> {
   defaultValue?: TValue | null;
   multiple?: false;
-  onChange?: (value: TValue | null) => void;
+  onChange?: (
+    e: React.MouseEvent | React.KeyboardEvent | React.FocusEvent | null,
```

Comment:
Is `React.SyntheticEvent` more suitable in this case? I got this from Material UI `Tabs`.

Reply:
Could be as well. Do you experience problems with the current implementation?

Reply:
Not yet. Seems like using union makes more sense because each type inherits from the `SyntheticEvent`.